internal/codegen/golang/go_type.go

```diff
@@ -1,50 +1,92 @@
 package golang
 
 import (
-	"github.com/kyleconroy/sqlc/internal/compiler"
-	"github.com/kyleconroy/sqlc/internal/config"
+	"github.com/kyleconroy/sqlc/internal/pattern"
+	"github.com/kyleconroy/sqlc/internal/plugin"
 )
 
-func goType(r *compiler.Result, col *compiler.Column, settings config.CombinedSettings) string {
+// XXX: dedupe
+func matchString(pat, target string) bool {
+	matcher, err := pattern.MatchCompile(pat)
+	if err != nil {
+		panic(err)
+	}
+	return matcher.MatchString(target)
+}
+
+func matches(o *plugin.Override, n *plugin.Identifier, defaultSchema string) bool {
```

Comment:
this is copied straight out of the python port - might be nice to stick somewhere shared

Reply:
i left a note re: deduplication here, i am happy to start an `sdk` package, but maybe that requires more thought for what the plugin api would actually look like